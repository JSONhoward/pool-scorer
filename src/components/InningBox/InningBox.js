import React, { useState, useRef } from 'react'
import styled from 'styled-components'
import { FaPlus, FaMinus } from 'react-icons/fa'
import anime from 'animejs/lib/anime.es.js'

const InningBoxStyledLeft = styled.div`
position: fixed;
top: 4rem;
left: ${props => props.px + 'px'};
height: 250px;
width: 100px;
background-color: white;
border-bottom-right-radius: 10px;
border: 1px solid black;
margin: auto 0;
`

const InningBoxStyledRight = styled.div`
position: fixed;
top: 4rem;
right: ${props => props.px + 'px'};
height: 250px;
width: 100px;
background-color: white;
border-bottom-left-radius: 10px;
border: 1px solid black;
margin: auto 0;
`

const TabLeft = styled.div`
position: relative;
display: grid;
place-items: center;
right: -99px;
height: 50px;
width: 50px;
color: black;
background-color: white;
border-top-right-radius: 10px;
border-bottom-right-radius: 10px;
cursor: pointer;

p {
position: absolute;
top: 0;
font-size: small;
}
`

const TabRight = styled.div`
position: relative;
display: grid;
place-items: center;
left: -50px;
height: 50px;
width: 50px;
color: black;
background-color: white;
border-top-left-radius: 10px;
border-bottom-left-radius: 10px;
cursor: pointer;

p {
position: absolute;
top: 0;
font-size: small;
}
`

const Table = styled.table`
position: relative;
top: 75px;
left: 50%;
height: 250px;
width: 100px;
transform: translate(-50%, -50%);
color: black;
font-size: 1rem;
border: 1px solid black;
border-spacing: 0px;
text-align: center;
border-Bottom-Right-Radius: ${props => props.left ? '10px' : '0px'};
border-Bottom-Left-Radius: ${props => props.left ? '0px' : '10px'};

th, tr, td {
    border: 1px solid black;
    border-collapse: collapse;
}
`

const InningBox = ({ players, scores }) => {
    const [isOpenLeft, setIsOpenLeft] = useState(false)
    const [isOpenRight, setIsOpenRight] = useState(false)
    const [pixelsLeft, setPixelsLeft] = useState(-100)
    const [pixelsRight, setPixelsRight] = useState(-100)

    const inningsTab = useRef()

    const openLeft = () => {
        let obj = {
            px: -100
        }

        setIsOpenLeft(true)
        anime({
            targets: obj,
            px: 0,
            autoplay: false,
            loop: false,
            duration: 1000,
            easing: 'easeInOutSine',
            update: () => setPixelsLeft(obj.px)
        }).play()
    }

    const closeLeft = () => {
        let obj = {
            px: 0
        }

        setIsOpenLeft(false)
        anime({
            targets: obj,
            px: -100,
            autoplay: false,
            loop: false,
            duration: 1000,
            easing: 'easeInOutSine',
            update: () => setPixelsLeft(obj.px)
        }).play()
    }

    const openRight = () => {
        let obj = {
            px: -100
        }

        setIsOpenRight(true)
        anime({
            targets: obj,
            px: 0,
            autoplay: false,
            loop: false,
            duration: 1000,
            easing: 'easeInOutSine',
            update: () => setPixelsRight(obj.px)
        }).play()
    }

    const closeRight = () => {
        let obj = {
            px: 0
        }

        setIsOpenRight(false)
        anime({
            targets: obj,
            px: -100,
            autoplay: false,
            loop: false,
            duration: 1000,
            easing: 'easeInOutSine',
            update: () => setPixelsRight(obj.px)
        }).play()
    }

    const inningList1 = scores[0].map((el, i) => {
        return (
            <tr key={i + 1}>
                <td key={i + 2}>{el}</td>
                <td key={i + 3} style={(i === 9) ? { borderBottomRightRadius: '10px' } : {}}>{i + 1}</td>
            </tr>
        )
    })

    const inningList2 = scores[1].map((el, i) => {
        return (
            <tr key={i + 1}>
                <td key={i + 2} style={(i === 9) ? { borderBottomLeftRadius: '10px' } : {}}>{i + 1}</td>
                <td key={i + 3}>{el}</td>
            </tr>
        )
    })

    return (
        <>
            {
                players === 1 ? (
                    <InningBoxStyledLeft px={pixelsLeft} ref={inningsTab}>
                        <TabLeft onClick={isOpenLeft ? closeLeft : openLeft}>
                            <p>Innings</p>
                            {
                                isOpenLeft ? <FaMinus /> : <FaPlus />
                            }
                        </TabLeft>
                        <Table left={true}>
                            <tbody>
                                <tr>
                                    <th>Score</th>
                                    <th>Inning</th>
                                </tr>
                                {inningList1}
                            </tbody>
                        </Table>
                    </InningBoxStyledLeft>
                )
                    :
                    (
                        <>
                            <InningBoxStyledLeft px={pixelsLeft} ref={inningsTab}>
                                <TabLeft onClick={isOpenLeft ? closeLeft : openLeft}>
                                    <p>Innings</p>
                                    {
                                        isOpenLeft ? <FaMinus /> : <FaPlus />
                                    }
                                </TabLeft>
                                <Table left={true}>
                                    <tbody>
                                        <tr>
                                            <th>Score</th>
                                            <th>Inning</th>
                                        </tr>
                                        {inningList1}
                                    </tbody>
                                </Table>
                            </InningBoxStyledLeft>
                            <InningBoxStyledRight px={pixelsRight} ref={inningsTab}>
                                <TabRight onClick={isOpenRight ? closeRight : openRight}>
                                    <p>Innings</p>
                                    {
                                        isOpenRight ? <FaMinus /> : <FaPlus />
                                    }
                                </TabRight>
                                <Table left={false}>
                                    <tbody>
                                        <tr>
                                            <th>Inning</th>
                                            <th>Score</th>
                                        </tr>
                                        {inningList2}
                                    </tbody>
                                </Table>
                            </InningBoxStyledRight>
                        </>
                    )
            }
        </>
    )
}

export default InningBox
