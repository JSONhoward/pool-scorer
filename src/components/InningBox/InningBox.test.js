import React from 'react'
import {create} from 'react-test-renderer'
import InningBox from './InningBox'

describe('InningBox Tests', () => {
    test('Snapshot Test', () => {
        const tree = create(<InningBox scores={[[null], [null]]} />).toJSON()

        expect(tree).toMatchSnapshot()
    })

    test('Receives Props', () => {
        const props = {
            scores: [[null], [null]],
            players: 1,
            top: '2rem'
        }
        const component = create(<InningBox {...props} />).root

        expect(component.props).toEqual(props)
    })
})