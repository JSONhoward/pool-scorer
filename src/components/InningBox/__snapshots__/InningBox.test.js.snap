// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`InningBox Tests Snapshot Test 1`] = `
Array [
  <div
    className="sc-AxjAm eHbWAW"
  >
    <div
      className="sc-AxiKw jckrDE"
      onClick={[Function]}
    >
      <p>
        Innings
      </p>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        />
      </svg>
    </div>
    <table
      className="sc-AxhUy dvlFGg"
    >
      <tbody>
        <tr>
          <th>
            Score
          </th>
          <th>
            Inning
          </th>
        </tr>
        <tr>
          <td />
          <td
            style={Object {}}
          >
            1
          </td>
        </tr>
      </tbody>
    </table>
  </div>,
  <div
    className="sc-AxirZ ggEqld"
  >
    <div
      className="sc-AxhCb fEigVe"
      onClick={[Function]}
    >
      <p>
        Innings
      </p>
      <svg
        fill="currentColor"
        height="1em"
        stroke="currentColor"
        strokeWidth="0"
        style={
          Object {
            "color": undefined,
          }
        }
        viewBox="0 0 448 512"
        width="1em"
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          d="M416 208H272V64c0-17.67-14.33-32-32-32h-32c-17.67 0-32 14.33-32 32v144H32c-17.67 0-32 14.33-32 32v32c0 17.67 14.33 32 32 32h144v144c0 17.67 14.33 32 32 32h32c17.67 0 32-14.33 32-32V304h144c17.67 0 32-14.33 32-32v-32c0-17.67-14.33-32-32-32z"
        />
      </svg>
    </div>
    <table
      className="sc-AxhUy bzYzLc"
    >
      <tbody>
        <tr>
          <th>
            Inning
          </th>
          <th>
            Score
          </th>
        </tr>
        <tr>
          <td
            style={Object {}}
          >
            1
          </td>
          <td />
        </tr>
      </tbody>
    </table>
  </div>,
]
`;
